src/storage/src/hummock/compactor/compactor_runner.rs

```diff
@@ -818,15 +819,16 @@ where
             // information about whether a key is deleted by a delete range in
             // the same SST. Therefore we need to construct a corresponding
             // delete key to represent this.
-            iter_key.epoch = earliest_range_delete_which_can_see_iter_key;
+            iter_key.epoch_with_gap =
+                EpochWithGap::new_from_epoch(earliest_range_delete_which_can_see_iter_key);
             sst_builder
                 .add_full_key(iter_key, HummockValue::Delete, is_new_user_key)
                 .verbose_instrument_await("add_full_key_delete")
                 .await?;
             last_table_stats.total_key_count += 1;
             last_table_stats.total_key_size += iter_key.encoded_len() as i64;
             last_table_stats.total_value_size += 1;
-            iter_key.epoch = epoch;
+            iter_key.epoch_with_gap = EpochWithGap::new_from_epoch(epoch);
```

Comment:
It seems that the original logic is to save the original epoch, set a new epoch and add the key, and then reset to the original epoch. The logic here will remove the gap in the original epoch.